raft/src/main/java/org/apache/kafka/raft/KafkaRaftClient.java

```diff
@@ -2132,15 +2132,32 @@ private long pollLeader(long currentTimeMs) {
             currentTimeMs
         );
 
-        long timeUntilSend = maybeSendRequests(
-            currentTimeMs,
-            partitionState
-                .lastVoterSet()
-                .voterNodes(state.nonAcknowledgingVoters().stream(), channel.listenerName()),
-            this::buildBeginQuorumEpochRequest
-        );
+        long timeUntilNextBeginQuorumSend = state.timeUntilBeginQuorumEpochTimerExpires(currentTimeMs);
+
+        if (timeUntilNextBeginQuorumSend == 0) {
+            timeUntilNextBeginQuorumSend = maybeSendRequests(
+                currentTimeMs,
+                partitionState
+                    .lastVoterSet()
+                    .voterNodes(channel.listenerName()),
```

Comment:
The leader should not send a BEGIN_QUORUM_EPOCH request to itself. Please write a detail test that checks this.